docs/api/public_php_api_search.md

```diff
@@ -81,10 +87,140 @@ As such, `query` is recommended when the search is based on user input.
+        }
+
+        return self::SUCCESS;
+    }
+}
+```
+
+Notice that the total number of items is retrieved differently for `ContentList` and `LocationList`. 
+
+!!! caution
+
+    The total count is the total number of matched items, regardless of pagination settings.
+    
+!!! tip "Repository filtering is SiteAccess-aware"
+
+    Repository filtering is SiteAccess-aware, which means you can skip the second argument of the `find` methods.
+    In that case languages from a current context are injected and added as a LanguageCode Criterion filter. 
+    
+You can use the following methods of the Filter:
+
+- `withCriterion` - add the first Criterion to the Filter
+- `andWithCriterion` - add another Criterion to the Filter using a LogicalAnd operation. If this is the first Criterion, this method works like `withCriterion`
+- `orWithCriterion` - add a Criterion using a LogicalOr operation. If this is the first Criterion, this method works like `withCriterion`
+- `withSortClause` - add a Sort Clause to the Filter
+- `sliceBy` - set limit and offset for pagination
+- `reset` - remove all Criteria, Sort Clauses, and pagination settings
+
+The following example filters for Folder Content items under the parent Location 2,
+sorts them by publication date and returns 10 results, starting from the third one:
+
+``` php
+$filter = new Filter();
+$filter
+    ->withCriterion(new Criterion\ContentTypeIdentifier('folder'))
+    ->andWithCriterion(new Criterion\ParentLocationId(2))
+    ->withSortClause(new SortClause\DatePublished(Query::SORT_ASC))
+    ->sliceBy(10, 2);
+```
+
+!!! note "Search Criteria and Sort Clause availability"
+
+    Not all Search Criteria and Sort Clauses are available for use in Repository filtering.
+    
+    Only Criteria implementing [`FilteringCriterion`](https://github.com/ezsystems/ezplatform-kernel/blob/v1.1.0/eZ/Publish/SPI/Repository/Values/Filter/FilteringCriterion.php)
+    and Sort Clauses implementing [`FilteringSortClause`](https://github.com/ezsystems/ezplatform-kernel/blob/v1.1.0/eZ/Publish/SPI/Repository/Values/Filter/FilteringSortClause.php)
+    are supported.
+
+    See to [Search Criteria](../guide/search/search_criteria_reference.md)
+    and [Sort Clause reference](../guide/search/sort_clause_reference.md) for details.
+    
+!!! tip
+
+    It is recommended to use an IDE that can recognize type hints when working with Repository Filtering.
+    If you try to use an unsupported Criterion or Sort Clause, the IDE will indicate an issue. 
+
 ## Searching in a controller
 
-You can use the `SearchService` similarly in a controller, as long as you provide the required parameters.
-For example, in the code below `locationId` is provided to list all children of a Location.
+You can use the `SearchService` or Repository filtering in a controller, as long as you provide the required parameters.
+For example, in the code below `locationId` is provided to list all children of a Location using the `SearchService`.
```

Comment:
```suggestion
For example, in the code below, `locationId` is provided to list all children of a Location using the `SearchService`.
```
I guess "using" does not mean "that uses" but ""by using" or "with"? I'd remove the ambiguity.